### PR TITLE
Ember CLI Addon Support

### PR DESCRIPTION
**Pull request by AndrewJo**
This PR adds much needed support for Ember CLI Addon projects that wish to use Compass compilers. Once merged in, the minor version should be bumped (0.3.2 → 0.4.0) and published to NPM.

## Major Changes

- Input tree for the Broccoli Compass Compiler has changed: `[inputPath]` → `[tree]`.

  This ensures that `ember-cli-compass-compiler` correctly checks for files in `addon/styles`.

- Adds `getSassDir` function to correctly search the input paths

  Without this, the compiler throws an ENOENT error about missing files/directory.

See #47.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -34,10 +35,33 @@ CompassCompilerPlugin.prototype.toTree = function(tree, inputPath, outputPath, i
...
+          var result = glob.sync(pattern);
+
+          // Found the file we're looking for
+          if (result.length > 0) {
+            return path.dirname(result[0]);
+          }
+
+          return '';
+        }, '');
+      }
```
Comment (quaertym) [inline index.js]: Need `;` here.

File: index.js [hunk 2]
```diff
@@ -48,17 +72,18 @@ CompassCompilerPlugin.prototype.toTree = function(tree, inputPath, outputPath, i
     var outputFileName = path.basename(outputPaths[file], extension) + extension; // new name for asset
     var outputDir = removeLeadingSlash(path.dirname(outputPaths[file])); // new directory for asset
     var node = new Funnel(compassTree, {
-      srcDir: outputPath,
```
Comment (quaertym) [inline index.js]: This change breaks my app. Why was that necessary?

File: index.js [hunk 3]
```diff
@@ -34,10 +35,33 @@ CompassCompilerPlugin.prototype.toTree = function(tree, inputPath, outputPath, i
...
-  var trees = Object.keys(outputPaths).reduce(function(trees, file) {
+  var trees = Object.keys(outputPaths).map(function(file) {
 
     // Watch inputTrees and compassOptions.importPath directories
-    var inputTrees = [inputPath];
+    var inputTrees = [tree];
+
+    // Define getSassDir function if not overridden by user
+    if (!compassOptions.getSassDir) {
+      compassOptions.getSassDir = function(inputTrees, inputPaths) {
```
Comment (quaertym) [inline index.js]: Let's move this outside of map function and better put in default options.

File: index.js (hunk 1 above)
Comment (AndrewJo) [inline index.js]: Oops! My bad. Will fix.

File: index.js (hunk 3 above)
Comment (AndrewJo) [inline index.js]: Yes. It's one of the reasons why I wanted to make the changes in the `broccoli-compass-compiler` in the beginning.

File: index.js (hunk 2 above)
Comment (AndrewJo) [inline index.js]: Because `srcDir` should be the source directory of the Compass compiler's output tree, not the `outputPath` within the compass output tree that may or may not exist.

If `srcDir` is set to `outputPath`, Ember CLI throws a `ENOENT` error when trying to compile `addon.scss` because the intermediate Compass output file actually lives at `COMPASS_OUTPUT_TREE/addon.css` instead of `COMPASS_OUTPUT_TREE/assets/addon.css`.

This goes the same for the test dummy app: `COMPASS_OUTPUT_TREE/app.css` and `COMPASS_OUTPUT_TREE/test-support.css`, **not:** `COMPASS_OUTPUT_TREE/asset/app.css` and `COMPASS_OUTPUT_TREE/asset/test-support.css`.

Can you tell me how it breaks your app and also the test environment (Ember CLI version, Node.js version, etc.)?

File: index.js (hunk 3 above)
Comment (quaertym) [inline index.js]: I think it is fine to override `getSassDir` function.

File: index.js (hunk 2 above)
Comment (quaertym) [inline index.js]: I wonder if this is a bug of Funnel, but removing `srcDir` definitely breaks my ember-cli app.

Comment (AndrewJo) [inline index.js]: Can you tell me how it breaks your app? Does Ember CLI throw an error?

Comment (AndrewJo) [inline index.js]: So I think the problem is that in regular Ember app projects, the intermediate Compass output file is actually `COMPASS_OUTPUT_TREE/assets/app.css` but for Ember addons, it lives at `COMPASS_OUTPUT_TREE/app.css` for some reason.

Comment (quaertym) [inline index.js]: @AndrewJo  Yep, that's right.

Comment (quaertym) [inline index.js]: What is the `outputPath` for addons?

Comment (AndrewJo) [inline index.js]: Oh my goodness, I'm an idiot. You're actually right. It was a problem with my local copy. I'll add `srcDir` back in.

**Comment by AndrewJo** [comment]
@quaertym I made the changes! Let me know if it works for your app.

**Comment by AndrewJo** [comment]
@quaertym Do you have any pushbacks for this PR? I verified this works with both normal Ember CLI App projects as well as Addon projects.

**Comment by quaertym** [comment]
@AndrewJo It has been a busy week. I will merge and publish ASAP.

**Comment by AndrewJo** [comment]
@quaertym Thanks! I've been busy myself. I appreciate your efforts to bring Compass compiler to Ember CLI! :+1:
